lib/matplotlib/patches.py

```diff
@@ -989,6 +989,78 @@ def set_path(self, path):
         self._path = path
 
 
+class StepPatch(PathPatch):
+    """An unclosed step path patch."""
+
+    @docstring.dedent_interpd
+    def __init__(self, values, edges, *,
+                 orientation='vertical', baseline=0, **kwargs):
+        """
+        Parameters
+        ----------
+        values : array-like
+            An array of y-values.
+
+        edges : array-like
+            A array of x-value edges, with ``len(edges) == len(vals) + 1``,
+            between which the curve takes on vals values.
+
+        orientation : {'vertical', 'horizontal'}, default: 'vertical'
+
+        baseline : float or None, default: 0
+            Determines starting value of the bounding edges or when
+            ``fill=True``, position of lower edge.
+
+        Other valid keyword arguments are:
+
+        %(Patch)s
+        """
+        self.baseline = baseline
+        self.orientation = orientation
+        self._edges = np.asarray(edges)
+        self._values = np.asarray(values)
+        verts, codes = self._update_data()
+        path = Path(verts, codes)
+        super().__init__(path, **kwargs)
+
+    def _update_data(self):
+        if np.isnan(np.sum(self._edges)):
+            raise ValueError('Nan values in "edges" are disallowed')
+        if self._edges.size - 1 != self._values.size:
+            raise ValueError('Size mismatch between "values" and "edges". '
+                             "Expected `len(values) + 1 == len(edges)`, but "
+                             "they are or lengths {} and {}".format(
+                                 self._edges.size, self._values.size)
+                             )
+        verts, codes = [], []
+        for idx0, idx1 in cbook.contiguous_regions(~np.isnan(self._values)):
+            x = np.repeat(self._edges[idx0:idx1+1], 2)
+            y = np.repeat(self._values[idx0:idx1], 2)
+            if self.baseline is not None:
+                y = np.hstack((self.baseline, y, self.baseline))
+            else:
+                y = np.hstack((y[0], y, y[-1]))
+            if self.orientation == 'vertical':
+                xy = np.column_stack([x, y])
+            else:
+                xy = np.column_stack([y, x])
+            verts.append(xy)
+            codes.append(np.array([Path.MOVETO] + [Path.LINETO]*(len(xy)-1)))
+        return np.vstack(verts), np.hstack(codes)
+
+    def set_edges(self, edges):
+        self._edges = np.asarray(edges)
+        verts, codes = self._update_data()
+        self.set_path(Path(verts, codes))
+        self.stale = True
+
+    def set_values(self, values):
```

Comment:
can these both have docstrings? I know the names are pretty explicit but when I just tried it out the very first thing I tried to do when using these was look at the docstrings via `shift-tab`

the `set_data` for an AxesImage (result of imshow) is this which I find fairly helpful:
```
Set the image array.

Note that this function does *not* update the normalization used.

Parameters
----------
A : array-like or `PIL.Image.Image`
```